### ISSUE TITLE
Blur Scenario when starting simulation from .exe file

### ISSUE THREAD

**Issue by MartinaB85** [lgsvl/simulator#1839]
Good morning,

I'm currently using the 2021.1 SVL version on Ubuntu 20.04.
When building a new binary of my scenario and starting the simulation through the Python script and the .exe file at the beginning everything is working fine but if I run it again the scenario becomes blur (see attached image).
This is also happened to my colleague.
Any idea why is it happening? Is it something related to the build process? Or because of the scenario we are using?

**Please note**: When building the binary I do not build the scenario asset bundle, it uploads the scenario directly from the cloud.

Thanks,

Martina

![Blur_Scenario_when_loading](https://user-images.githubusercontent.com/73334464/139012474-bf268059-21a9-4232-923e-595644b15a08.png)

**Comment by MartinaB85** [comment]
Yes, I was calling the reset in the code, but even if commenting that line and leaving the sim.load only (as you see in the picture in attached) same things happens. I really would need to stay on this version right now, as changing SVL version implies doing again the scenario and adjusting all the code in the controller since some functions are not the same.

When opening the .exe if I choose go Online it also does not work. So maybe the issue is related to this? That somehow it can not go online and load the scenario? 

Any ideas on which could be the issue?


Thanks,

Martina

![Screenshot from 2021-11-04 15-56-09](https://user-images.githubusercontent.com/73334464/140337295-d0484f66-2c1c-4b46-9013-861af0991bf5.png)

**Comment by EricBoiseLGSVL** [comment]
Thanks for posting the code, that helps.

You aren't checking to see if the map is already loaded.  Please look at all the quick script examples we provide to see the check for environment load.  If the map is already loaded, it needs to call reset, not load again.

**Comment by EricBoiseLGSVL** [comment]
Oh, so the issue is happening with the lines un-commented out?  Sorry, I was confused on the example posted.
You said you are building the binary.  What changes are you making to it?  Does this happen with the release binary we provide?

**Comment by MartinaB85** [comment]
The issue is happening regardless. The same issue happens building the binary from your original simulator, without changes in the code. After building the binary it works fine for a while then it gets blur..so maybe is it something in the build process? Or any settings which might be wrong?

Here in attached the warnings I get while building the binary of my project. Not sure if it can help you detecting the issue, hopefully yes.

![Build_Console](https://user-images.githubusercontent.com/73334464/140938053-8b4e2640-677b-4cbd-a3b1-a4bf2d6d9a1a.png)

**Comment by EricBoiseLGSVL** [comment]
Thanks for the log, it always helps, but those warnings are fine.
I'm confused, do you have this issue with the binary we provide?
Or is it only when you make a binary yourself?
I ask because we haven't been able to reproduce so it might be a hardware or driver issue.

Let's look into the playerlog next, please post here.
Also, what hardware, drivers are you using? Nvidia smi will help

**Comment by MartinaB85** [comment]
Yes, I do have the same issue with your binary as well. 
I was searching my whole PC but I couldn't find any Player.log. And the option Use Player Log is checked in player settings. So not sure why it is not being created.
Regarding my pc, I have attached a pic with the properties along with the output of the nvidia-smi.

![PCSettings](https://user-images.githubusercontent.com/73334464/141134066-54fc74b9-ee0e-4f90-ab71-60638e5bbcb3.png)
![Nvidia_smi_Output](https://user-images.githubusercontent.com/73334464/141134150-ed09ae4f-a9c7-4def-8623-f54e41a2cda7.png)

**Comment by MartinaB85** [comment]
Yes, I do have the same issue with your binary as well.
In attached the Player.log file.
Regarding my pc, I have attached a pic with the properties along with the output of the nvidia-smi.

[Player.log](https://github.com/lgsvl/simulator/files/7513762/Player.log)
![PCSettings](https://user-images.githubusercontent.com/73334464/141139572-74bee4d9-f762-4833-b050-07b3b85f47f5.png)
![Nvidia_smi_Output](https://user-images.githubusercontent.com/73334464/141139588-df71f17e-5c89-4b46-90cf-4d33e8fa199f.png)

**Comment by EricBoiseLGSVL** [comment]
Thanks for the info this helps.  I see a couple of things:
- Looks like the sim is loading ASUS as the gpu.  Please make sure simulator is using that 3080 beast you have :)
- Looks like you have mesa drivers installed.  Please uninstall.  Unity will have all kinds of graphical issues with those.
- I see some new methods in VehicleController.cs and many null errors.  Please make sure no null errors are happening, check the console panel in unity while developing and in the info panel in simulator.
- Once these are cleared up, let us know if you are still having issues.

**Comment by MartinaB85** [comment]
Thank you very much for helping me.
I will try to fix these issues..just a question on the first point, how can I make sure the simulator is using my graphic cards? Are there any settings I need to adjust in Unity in order to "force" it to use the NVIDIA 3080?

**Comment by EricBoiseLGSVL** [comment]
yes, in windows you can use nivida control panel to assign gpu to an application.  I'm sure there is a way on Linux.  @lemketron ?

**Comment by MartinaB85** [comment]
Good morning, I have checked all the things you suggested me, but I'm still experiencing the same issue.
I share the player.log for the case when the scene is clearly visible and for the case when the scene is blurred.
I do not detect particular issues..but maybe you can double check.
Are there any other things I should check to understand why this is happening? 

Thanks,

Martina

[Player_Visible_Scene.log](https://github.com/lgsvl/simulator/files/7639918/Player_Visible_Scene.log)
[Player_NOT_Visible_Scene.log](https://github.com/lgsvl/simulator/files/7639920/Player_NOT_Visible_Scene.log)

**Comment by EricBoiseLGSVL** [comment]
It seems from the player log, thanks for posting, that you are getting many errors when trying to run again.  Since this is an older version, I recommend you move to the latest that fixes many issues.

You can try to fix the listed issues below if you can't update.

I see you are using a terrain, that is not supported in 2021.1
```
TerrainCollider: MeshCollider is not supported on terrain at the moment. 
```

You are missing a nav mesh at the spawn location
```
Brennero missing NavMesh at (321.5, 0.0, 221.8) please create navmesh at this point. Pedestrian manager disabled
```

Not sure what is going on here but it seems a texture problem
```
Compute dispatch: missing UAV ID %d 
(Filename:  Line: 0)

Compute dispatch: missing texture ID %d 
(Filename:  Line: 0)

Compute dispatch: missing texture ID %d 
(Filename:  Line: 0)

Compute dispatch: missing texture ID %d 
(Filename:  Line: 0)

Compute dispatch: missing UAV ID %d 
(Filename:  Line: 0)

Compute dispatch: missing texture ID %d 
(Filename:  Line: 0)

Compute dispatch: missing texture ID %d 
(Filename:  Line: 0)
```

Seems like a custom visualizer or a reference is broken?
```
NullReferenceException: Object reference not set to an instance of an object.  Is this a custom sensor with a camera base?
  at Simulator.Sensors.CameraSensorBase.OnVisualize (Simulator.Sensors.UI.Visualizer visualizer) [0x00007] in <42c6685c778249ae9ddfa377e3b129aa>:0 
  at Simulator.Sensors.UI.Visualizer.LateUpdate () [0x00006] in <42c6685c778249ae9ddfa377e3b129aa>:0 
```

**Comment by MartinaB85** [comment]
These errors regarding the texture are happening if:
1) I do the build of the simulator (the first time everything's loading fine and everything works).
2) I change whatever small things in the code and I run the simulation in the Unity Editor.
3) I reopen the build made before the changes and I get the blur scenario and missing textures.

I tried to delete the library folder, open Unity again and do the build again, including in Resources all the textures I am using but the issue still persists.

Is it something normal that you do the build and then you run the project again in the editor changing a variable and then you want to reopen the previous build but it doesn't load? I mean, in my mind it shouldn't happen because if you do a build then it should work whenever you open it, no matter if you changed something in the source code of the project which is being run in the editor, right?

Is it something that ever happened to you? 

Many thanks,

Martina

**Comment by EricBoiseLGSVL** [comment]
What do you mean for 2? What are you changing?  If you are altering cached assets with changes in the Editor, you will break those assets in the binary without those changes.  Understanding what you are exactly doing in Editor will help solve this.  Why are you making changes to source code in editor and then running a binary without those changes?

**Comment by MartinaB85** [comment]
I have a string variable in the NPCBehaviour script as well as in the vehicleController script. This string needs to be changed to simulate 4 different cases. So I change the string to value1 and build with Name1, then change it to value2 and build with Name2 and so on for four times. If I run all the four builds everything's fine. But if I see something wrong in the behaviour of one of these builds I need to go back to the editor and fix things. And if for some reason I want to check out the previous build (not in the editor) it doesn't load if I made changes to the code. If I do a new build of one of them then all four (even the previous ones) are again working fine.
So just for my understanding, this is something normal right?

**Comment by EricBoiseLGSVL** [comment]
Yes this is as intended. If you make changes in source and don't build a new binary, you will have errors or other issues that will prevent loading.  That is why we provided development debug so you can test local changes before you bring it to a binary.

Why not make an api command to swap logic.  I am confused on why you need 4 separate builds.  I understand changing logic needs new builds but what is the significance of the name and string.  This seems like a good place to add functionality on addagent with strings provided as parameters.

**Comment by MartinaB85** [comment]
I see, thanks for your clarification.! Now I know that is "normal".
